[torch] Implement c10::BFloat16 ctor from __hip_bfloat16

danzimm opened the pull request:
Summary: Pretty straightfoward. ROCm 6.2.0 changed the `__hip_bfloat16` API (see [this PR](https://github.com/ROCm/clr/commit/481912a1fd6b8c400347688663820aa2b27df0ea)), so we gate impl on `__BF16_HOST_DEVICE__` macro to support older and newer versions of ROCm.

Test Plan: CI

Differential Revision: D60024830

houseroad reviewed (commented):
Could you sign up the EasyCLA and trigger other CIs?

houseroad commented:
Click the EasyCLA link above and complete the agreement? cc: @danzimm

danzimm commented:
@houseroad I signed the EasyCLA, guess it didn't retrigger that check. Lemme repush to see if that does the trick, or if there's something I'm doing wrong still

danzimm committed:
[torch] Implement c10::BFloat16 ctor from __hip_bfloat16 (#131359)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/131359

Pretty straightfoward. ROCm 6.2.0 changed the `__hip_bfloat16` API (see [this PR](https://github.com/ROCm/clr/commit/481912a1fd6b8c400347688663820aa2b27df0ea)), so we gate impl on `__BF16_HOST_DEVICE__` macro to support older and newer versions of ROCm.

Test Plan: CI

Reviewed By: houseroad

Differential Revision: D60024830

houseroad reviewed (approved):
Looks good.
